### PR TITLE
Project Mongo API - Jessika Hage

### PR DESCRIPTION
**Pull request by jessika-hage**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,39 +1,175 @@
...
+// First page that shows all the endpoints
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.send(listEndpoints(app))
+});
+
+// Function for different queries
+// Global scope to be used in all, movies and tvshows routes
+const queries = (title, genre, releaseYear, country) => {
```
Comment (jesinkey) [inline server.js:55]: Nice sorting of querys!

File: server.js [hunk 2]
```diff
@@ -1,39 +1,175 @@
...
+// Route for all titles
+app.get('/all', async (req, res) => {
+  const { title, genre, releaseYear, country, page, sort } = req.query;
+  const titleRegex = new RegExp(title, 'i');
+  const genreRegex = new RegExp(genre, 'i');
+  const countryRegex = new RegExp(country, 'i');
+
+  // Find results on all titles
+  // Search by queries, sort on releaseYear and limit to 20 per page.
+  const data = await NetflixData.find(queries(titleRegex, genreRegex, releaseYear, countryRegex))
```
Comment (jesinkey) [inline server.js:97]: Awesome additions of extra sorting and limiting pages!

File: server.js [hunk 3]
```diff
@@ -1,39 +1,175 @@
...
+    return data.length > 0 
+      ? res.json(data) 
+      : res.json({ result: "Nothing found on that query, try again!" });
+  } catch (error) {
+    res.status(400).json({ error: 'Something went wrong!', details: error })
+  }
+});
+
+// Route for movies
+app.get('/movies', async (req, res) => {
```
Comment (jesinkey) [inline server.js:112]: Nice job creating a split between tv-shows and movies!

**Review by jesinkey — APPROVED** [review]
Great job! The code was easy to follow and you tried a lot of new different things.

File: README.md [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
+* /all - lists all movies and tvshows. Available queries are title, genre, releaseYear, country, sort and page. You can sort on releaseyear and there is a limit on 20 results per page. 
+* /movies - lists all movies and available queries are sort and page (same as above).
+* /tvshows - lists all tvshow and available queries are sort and page (same as above).
+* /all/:id - param to search for id.
+
+Endpoint examples: 
+https://move-times.herokuapp.com/all?sort=oldest&page=2
+https://move-times.herokuapp.com/movies?country=brazil&page=1
+https://move-times.herokuapp.com/tvshows?genre=drama&sort=newest
+
```
Comment (KarolCislowski) [inline README.md:25]: Nice documentation 👍🏻

File: server.js [hunk 5]
```diff
@@ -1,39 +1,175 @@
...
+    query.listed_in = genre;
+  }
+  if (releaseYear) {
+    query.release_year = releaseYear;
+  }
+  if (country) {
+    query.country = country;
+  }
+  return query;
+};
```
Comment (KarolCislowski) [inline server.js:70]: nice filtering 👍🏻

File: server.js [hunk 6]
```diff
@@ -1,39 +1,175 @@
...
+
+// Function to sort on newest and oldest movies
+// Global scope to be used in all, movies and tvshows routes
+const sortOnYear = (sort) => {
+  if (sort === 'newest') { 
+    return { release_year: -1 } 
+  } else if (sort === 'oldest') { 
+    return { release_year: 1 }
+  }
+};
```
Comment (KarolCislowski) [inline server.js:80]: nice sorting 👍🏻

File: server.js [hunk 7]
```diff
@@ -1,39 +1,175 @@
...
+
+// Function for pages, skipping 20 per page
+// Global scope to be used in all, movies and tvshows routes
+const pageResults = (page) => {
+  return ((page - 1) * 20)
+};
```
Comment (KarolCislowski) [inline server.js:86]: ❤️

**Review by KarolCislowski — APPROVED** [review]
Great API 👍🏻

**Pull request closed without merging**